backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/predicate/impl/ElasticsearchBooleanJunctionPredicateBuilder.java

```diff
@@ -124,6 +129,10 @@ protected JsonObject doBuild(ElasticsearchSearchPredicateContext context,
 		return outerObject;
 	}
 
+	private boolean hasAtLeastOneMustPredicate() {
+		return mustClauseBuilders != null || mustNotClauseBuilders != null;
```

Comment:
From the Elasticsearch 6.7 documentation:

> If the bool query is in a filter context or has neither must or filter then at least one of the should queries must match a document for it to match the bool query. This behavior may be explicitly controlled by setting the minimum_should_match parameter.

https://www.elastic.co/guide/en/elasticsearch/reference/6.7/query-dsl-bool-query.html

The documentation mentions `must` and `filter`, but your condition mentions `must` and `mustNot`. I wonder if it's accurate? Maybe you should add tests (in the same test method) with "one should, one filter" and "one should, one mustNot" in a filter context, just to be sure?

Reply:
Good idea. Let's add the test so that to verify whether  `|| mustNotClauseBuilders != null` is really need.

Reply:
You were right, according to the new test. A must-not is treated differently from a must.

Reply:
And what about "filter"? Or maybe you're still working on it?

Reply:
Sorry, I forgot it.
I think I've addressed it now.